src/plugins/auto/src/plugin.cpp

```diff
@@ -911,39 +911,71 @@ std::vector<DeviceInformation> Plugin::filter_device(const std::vector<DeviceInf
 }
 
 std::vector<DeviceInformation> Plugin::filter_device_by_model(const std::vector<DeviceInformation>& meta_devices,
-                                                                const std::shared_ptr<const ov::Model>& model) const {
+                                                              const std::shared_ptr<const ov::Model>& model,
+                                                              PluginConfig& load_config) const {
     if (meta_devices.empty()) {
         OPENVINO_THROW("No available device to filter ", get_device_name(), " plugin");
     }
 
     std::vector<DeviceInformation> filter_device;
-    auto is_stateful = [&]() {
-        for (auto& op : model->get_ops()) {
-            if (std::dynamic_pointer_cast<ngraph::op::AssignBase>(op) ||
-                std::dynamic_pointer_cast<ngraph::op::ReadValueBase>(op)) {
-                    LOG_INFO_TAG("stateful mode, try deployed to CPU");
-                    return true;
-                }
-        }
-        return false;
-    };
+    std::vector<std::string> stateful_node_names;
 
     // Check if CPU is in candidate list
     auto cpuiter = std::find_if(meta_devices.begin(), meta_devices.end(), [](const DeviceInformation& device_info) {
         return device_info.device_name.find("CPU") != std::string::npos;
     });
-
     // If CPU is in candidate list, load dynamic model to CPU first
     // For MULTI do not only load stateful model to CPU
     // For AUTO CTPUT only load stateful model to CPU
-    if (((model->is_dynamic()) || (is_stateful() && get_device_name() != "MULTI")) && cpuiter != meta_devices.end()) {
+    if (model->is_dynamic() && cpuiter != meta_devices.end()) {
         filter_device.push_back(*cpuiter);
         return filter_device;
     }
-
     // If CPU is not in candidate list, continue to run selection logic regardless of whether the input model is a
     // dynamic model or not
-    return meta_devices;
+
+    for (auto& op : model->get_ops()) {
+        if (std::dynamic_pointer_cast<ov::op::util::AssignBase>(op) ||
+            std::dynamic_pointer_cast<ov::op::util::ReadValueBase>(op)) {
+            LOG_INFO_TAG("stateful mode, try deployed to CPU");
+            stateful_node_names.push_back(op->get_friendly_name());
+        }
+    }
+    if (stateful_node_names.empty()) {
+        // not stateful model
+        return meta_devices;
+    }
+
+    // disable CPU_HELP if model is stateful
+    load_config.set_property(ov::intel_auto::enable_startup_fallback(false));
```

Comment:
We should disable ov::intel_auto::enable_runtime_fallback as well. TODO: enable enable_runtime_fallback when stateful reset_state is ready.

Reply:
Updated.